packages/contracts-bedrock/src/L2/SoulGasToken.sol

```diff
@@ -151,6 +154,26 @@ contract SoulGasToken is ERC20Upgradeable, OwnableUpgradeable {
         }
     }
 
+    /// @notice allowSgtValue is called by the owner to enable whitelist contracts to consume sgt as msg.value
+    function allowSgtValue(address[] calldata contracts) external onlyOwner {
+        require(IS_BACKED_BY_NATIVE, "allowSgtValue should only be called when IS_BACKED_BY_NATIVE");
+        SoulGasTokenStorage storage $ = _getSoulGasTokenStorage();
+        uint256 i;
+        for (i = 0; i < contracts.length; i++) {
+            $._allow_sgt_value[contracts[i]] = true;
+        }
```

Comment:
Should we emit events to track the historical events easier?

Reply:
Good point, done in 62f2ca073785cce7e94cc37065db4f858f06659e.